### PR TITLE
suite-nonsplit-guest-configured-bundles

### PR DESCRIPTION
**Pull request by himanshu-pal-metacube**
https://spryker.atlassian.net/browse/CC-16814 
Guest configured bundles

### CODE REVIEW

File: tests/api/suite_nonsplit/glue/configured_bundles_endpoints/guest_configured_bundles/positive.robot [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+Test Setup    TestSetup
+Resource    ../../../../../../resources/common/common_api.robot
+Default Tags    glue
+
+*** Test Cases ***
+ENABLER
+    TestSetup
+
+### POST ###
+# bug https://spryker.atlassian.net/browse/CC-19233
```
Comment (Katerynaa-Lavryk) [inline tests/api/suite_nonsplit/glue/configured_bundles_endpoints/guest_configured_bundles/positive.robot:12]: Please re-check if this bug is related to this test. This bug is about vouchers and the test is about configured bundle

File: tests/api/suite_nonsplit/glue/configured_bundles_endpoints/guest_configured_bundles/positive.robot [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+    And Response body parameter should be greater than:    [data][attributes][totals][priceToPay]    0
+    And Response should contain the array larger than a certain size:    [data][attributes][discounts]    0
+    And Response body parameter should not be EMPTY:    [data][attributes][discounts][0][displayName]
+    And Response body parameter should not be EMPTY:    [data][attributes][discounts][0][amount]
+    And Response body parameter should be:    [data][attributes][discounts][0][code]    None
+    And Response should contain the array smaller than a certain size:    [data][attributes][thresholds]    2
+    And Response body parameter should be:    [data][links][self]    ${glue_url}/guest-carts/${guest_cart_id}
+    [Teardown]    Run Keyword    Cleanup all items in the guest cart:    ${guest_cart_id}
+
+# bug https://spryker.atlassian.net/browse/CC-19233
```
Comment (Katerynaa-Lavryk) [inline tests/api/suite_nonsplit/glue/configured_bundles_endpoints/guest_configured_bundles/positive.robot:40]: Please re-check if this bug is related to this test and update the tests below if needed

File: tests/api/suite_nonsplit/glue/configured_bundles_endpoints/guest_configured_bundles/positive.robot (hunk 1 above)
Comment (himanshu-pal-metacube) [reply tests/api/suite_nonsplit/glue/configured_bundles_endpoints/guest_configured_bundles/positive.robot:12]: Yes, In response data we are getting discount attribute so, we have to write tests for it.

File: tests/api/suite_nonsplit/glue/configured_bundles_endpoints/guest_configured_bundles/positive.robot (hunk 2 above)
Comment (himanshu-pal-metacube) [reply tests/api/suite_nonsplit/glue/configured_bundles_endpoints/guest_configured_bundles/positive.robot:40]: Yes, In response data we are getting discount attribute so, we have to write tests for it.

File: tests/api/suite_nonsplit/glue/configured_bundles_endpoints/guest_configured_bundles/positive.robot (hunk 1 above)
Comment (OKrivtsova) [reply tests/api/suite_nonsplit/glue/configured_bundles_endpoints/guest_configured_bundles/positive.robot:12]: do did you write tests for it? I don't see any additional tests in commits, if you did please give the link

File: tests/api/suite_nonsplit/glue/configured_bundles_endpoints/guest_configured_bundles/positive.robot (hunk 2 above)
Comment (OKrivtsova) [reply tests/api/suite_nonsplit/glue/configured_bundles_endpoints/guest_configured_bundles/positive.robot:40]: do did you write tests for it? I don't see any additional tests in commits, if you did please give the link

File: tests/api/suite_nonsplit/glue/configured_bundles_endpoints/guest_configured_bundles/positive.robot [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+    And Each array element of array in response should contain value:    [included]    configuredBundleItem
+    And Response body parameter should be:    [included][0][attributes][configuredBundleItem][quantityPerSlot]    1
+    And Response body parameter should be:    [included][0][attributes][configuredBundleItem][slot][uuid]    ${configurable_bundle_slot_5_uuid}
+    And Response should contain the array of a certain size:    [included][0][attributes][selectedProductOptions]    0
+    [Teardown]    Run Keyword    Cleanup all items in the guest cart:    ${guest_cart_id}
+
+Add_configured_bundle_include_concrete_products
+    [Setup]    Run Keyword    I set Headers:    Content-Type=${default_header_content_type}    X-Anonymous-Customer-Unique-Id=${random}
+    When I send a POST request:    /guest-carts//guest-configured-bundles?include=concrete-products,guest-cart-items    {"data": {"type": "guest-configured-bundles","attributes": {"quantity": 1,"templateUuid": "${configurable_bundle_template_2_uuid}","items": [{"sku": "${configurable_bundle.slot_5.product_1}","quantity": 1,"slotUuid": "${configurable_bundle_slot_5_uuid}"}]}}}
```
Comment (OKrivtsova) [inline tests/api/suite_nonsplit/glue/configured_bundles_endpoints/guest_configured_bundles/positive.robot:152]: remove double //

File: tests/api/suite_nonsplit/glue/configured_bundles_endpoints/guest_configured_bundles/positive.robot [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+    And Each array element of array in response should contain value:    [included]    displayName
+    And Each array element of array in response should contain value:    [included]    isExclusive
+    And Each array element of array in response should contain value:    [included]    expirationDateTime
+    And Each array element of array in response should contain value:    [included]    discountPromotionAbstractSku
+    And Each array element of array in response should contain value:    [included]    discountPromotionQuantity
+    [Teardown]    Run Keyword    Cleanup all items in the guest cart:    ${guest_cart_id}
+
+Add_configured_bundle_include_guest_cart_items
+    [Setup]    Run Keyword    I set Headers:    Content-Type=${default_header_content_type}    X-Anonymous-Customer-Unique-Id=${random}
+    When I send a POST request:    /guest-carts//guest-configured-bundles?include=guest-cart-items    {"data": {"type": "guest-configured-bundles","attributes": {"quantity": 1,"templateUuid": "${configurable_bundle_template_2_uuid}","items": [{"sku": "${configurable_bundle.slot_5.product_1}","quantity": 1,"slotUuid": "${configurable_bundle_slot_5_uuid}"}]}}}
```
Comment (OKrivtsova) [inline tests/api/suite_nonsplit/glue/configured_bundles_endpoints/guest_configured_bundles/positive.robot:98]: remove double //

File: tests/api/suite_nonsplit/glue/configured_bundles_endpoints/guest_configured_bundles/positive.robot [hunk 5]
```diff
@@ -0,0 +1,300 @@
...
+    And Response should contain the array of a certain size:    [data][relationships][guest-cart-items][data]    2   
+    And Response should contain the array of a certain size:    [included]    2
+    And Each array element of array in response should contain nested property with value:    [included]    [attributes][sku]    ${configurable_bundle.slot_5.product_1}
+    And Each array element of array in response should contain nested property with value:    [included]    [attributes][quantity]    1
+    [Teardown]    Run Keyword    Cleanup all items in the guest cart:    ${guest_cart_id}
+
+Add_other_configured_bundle_product_with_same_template
+    [Setup]    Run Keyword    I set Headers:    Content-Type=${default_header_content_type}    X-Anonymous-Customer-Unique-Id=${random}
+    When I send a POST request:    /guest-carts//guest-configured-bundles    {"data": {"type": "guest-configured-bundles","attributes": {"quantity": 1,"templateUuid": "${configurable_bundle_template_2_uuid}","items": [{"sku": "${configurable_bundle.slot_5.product_1}","quantity": 1,"slotUuid": "${configurable_bundle_slot_5_uuid}"}]}}}
```
Comment (OKrivtsova) [inline tests/api/suite_nonsplit/glue/configured_bundles_endpoints/guest_configured_bundles/positive.robot:243]: remove double //

File: tests/api/suite_nonsplit/glue/configured_bundles_endpoints/guest_configured_bundles/positive.robot [hunk 6]
```diff
@@ -0,0 +1,300 @@
...
+    And Response body parameter should be:    [included][1][attributes][sku]    ${configurable_bundle.slot_5.product_2}
+    [Teardown]    Run Keyword    Cleanup all items in the guest cart:    ${guest_cart_id}
+
+
+
+### PATCH ###
+# bug https://spryker.atlassian.net/browse/CC-19233
+Update_configured_bundle_product_quantity  
+    [Setup]    Run Keywords    I set Headers:    Content-Type=${default_header_content_type}    X-Anonymous-Customer-Unique-Id=${random}
+    ...    AND    I send a POST request:    /guest-carts//guest-configured-bundles?include=guest-cart-items    {"data": {"type": "guest-configured-bundles","attributes": {"quantity": 1,"templateUuid": "${configurable_bundle_template_2_uuid}","items": [{"sku": "${configurable_bundle.slot_5.product_1}","quantity": 1,"slotUuid": "${configurable_bundle_slot_5_uuid}"}]}}}
```
Comment (OKrivtsova) [inline tests/api/suite_nonsplit/glue/configured_bundles_endpoints/guest_configured_bundles/positive.robot:264]: remove double //

File: tests/api/suite_nonsplit/glue/configured_bundles_endpoints/guest_configured_bundles/positive.robot [hunk 7]
```diff
@@ -0,0 +1,300 @@
...
+    And Response body parameter should be:    [data][links][self]    ${glue_url}/guest-carts/${guest_cart_id}
+    And Response body parameter should be:    [included][0][attributes][sku]    ${configurable_bundle.slot_5.product_1} 
+    And Response body parameter should be:    [included][0][attributes][quantity]    2
+    [Teardown]    Run Keyword    Cleanup all items in the guest cart:    ${guest_cart_id}
+
+
+### DELETE ###
+Delete_configured_bundle_from_cart 
+    [Setup]    Run Keywords    I set Headers:    Content-Type=${default_header_content_type}    X-Anonymous-Customer-Unique-Id=${random}
+    ...    AND    I send a POST request:    /guest-carts//guest-configured-bundles?include=guest-cart-items    {"data": {"type": "guest-configured-bundles","attributes": {"quantity": 1,"templateUuid": "${configurable_bundle_template_2_uuid}","items": [{"sku": "${configurable_bundle.slot_5.product_1}","quantity": 1,"slotUuid": "${configurable_bundle_slot_5_uuid}"}]}}}
```
Comment (OKrivtsova) [inline tests/api/suite_nonsplit/glue/configured_bundles_endpoints/guest_configured_bundles/positive.robot:295]: remove double //

File: tests/api/suite_nonsplit/glue/configured_bundles_endpoints/guest_configured_bundles/positive.robot (hunk 1 above)
Comment (himanshu-pal-metacube) [reply tests/api/suite_nonsplit/glue/configured_bundles_endpoints/guest_configured_bundles/positive.robot:12]: Removed the bug id as bug has been resolved.

File: tests/api/suite_nonsplit/glue/configured_bundles_endpoints/guest_configured_bundles/positive.robot (hunk 2 above)
Comment (himanshu-pal-metacube) [reply tests/api/suite_nonsplit/glue/configured_bundles_endpoints/guest_configured_bundles/positive.robot:40]: Removed the bug id as bug has been resolved.

File: tests/api/suite_nonsplit/glue/configured_bundles_endpoints/guest_configured_bundles/positive.robot (hunk 3 above)
Comment (himanshu-pal-metacube) [reply tests/api/suite_nonsplit/glue/configured_bundles_endpoints/guest_configured_bundles/positive.robot:152]: Removed

File: tests/api/suite_nonsplit/glue/configured_bundles_endpoints/guest_configured_bundles/positive.robot (hunk 4 above)
Comment (himanshu-pal-metacube) [reply tests/api/suite_nonsplit/glue/configured_bundles_endpoints/guest_configured_bundles/positive.robot:98]: Removed

File: tests/api/suite_nonsplit/glue/configured_bundles_endpoints/guest_configured_bundles/positive.robot (hunk 5 above)
Comment (himanshu-pal-metacube) [reply tests/api/suite_nonsplit/glue/configured_bundles_endpoints/guest_configured_bundles/positive.robot:243]: Removed

File: tests/api/suite_nonsplit/glue/configured_bundles_endpoints/guest_configured_bundles/positive.robot (hunk 6 above)
Comment (himanshu-pal-metacube) [reply tests/api/suite_nonsplit/glue/configured_bundles_endpoints/guest_configured_bundles/positive.robot:264]: Removed

File: tests/api/suite_nonsplit/glue/configured_bundles_endpoints/guest_configured_bundles/positive.robot (hunk 7 above)
Comment (himanshu-pal-metacube) [reply tests/api/suite_nonsplit/glue/configured_bundles_endpoints/guest_configured_bundles/positive.robot:295]: Removed

**Review by OKrivtsova — APPROVED** [review]

**Review by OKrivtsova — APPROVED** [review]

**Pull request merged**